repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/samlpolicy/SamlFlow10Test.groovy

```diff
@@ -267,4 +260,104 @@ class SamlFlow10Test extends ReposeValveTest {
+        given: "the origin service will return an Identity fault as JSON"
+        fakeIdentityV2Service.generateTokenFromSamlResponseHandler = { Request request, boolean shouldReturnXml ->
+            def values = [name: fault, code: code, message: message]
+            new DeproxyResponse(code, null, [(CONTENT_TYPE): APPLICATION_JSON], createIdentityFaultJsonWithValues(values))
+        }
+
+        when: "the request is sent to Repose asking for a JSON response"
+        def mc = sendSamlRequestForLegacyIssuer((ACCEPT): APPLICATION_JSON)
+
+        then: "the client receives the response code sent by the origin service"
+        mc.receivedResponse.code as Integer == code
+
+        when: "the fault response sent by the origin service is parsed as JSON"
+        def sentResponseJson = jsonSlurper.parseText(mc.handlings[0].response.body as String)
+
+        and: "the fault response received by the client is parsed as JSON"
+        def receivedResponseJson = jsonSlurper.parseText(mc.receivedResponse.body as String)
+
+        then: "the JSON response was not altered by Repose"
+        receivedResponseJson == sentResponseJson
+
+        where:
+        code                     | fault           | message
+        SC_BAD_REQUEST           | "badRequest"    | "Error code: 'FED-006'; Subject is not specified"
+        SC_INTERNAL_SERVER_ERROR | "identityFault" | "Service Unavailable"
+    }
+
+    @Unroll
+    def "a non-successful response code of #code from the origin service with content-type application/xml will be returned to the client unaltered"() {
+        given: "the origin service will return an Identity fault as XML"
+        fakeIdentityV2Service.generateTokenFromSamlResponseHandler = { Request request, boolean shouldReturnXml ->
+            def values = [name: fault, code: code, message: message]
+            new DeproxyResponse(code, null, [(CONTENT_TYPE): APPLICATION_XML], createIdentityFaultXmlWithValues(values))
+        }
+
+        when: "the request is sent to Repose asking for an XML response"
+        def mc = sendSamlRequestForLegacyIssuer((ACCEPT): APPLICATION_XML)
+
+        then: "the client receives the response code sent by the origin service"
+        mc.receivedResponse.code as Integer == code
+
+        and: "the XML sent by the origin service and received by the client are similar enough"
+        new Diff(mc.handlings[0].response.body as String, mc.receivedResponse.body as String).similar()
+
+        where:
+        code                     | fault           | message
+        SC_BAD_REQUEST           | "badRequest"    | "Saml response issueInstant cannot be in the future."
+        SC_INTERNAL_SERVER_ERROR | "identityFault" | "Internal Server Error"
+    }
+
+    def "an unsupported content-type from the origin service will be returned to the client unaltered"() {
```

Comment:
Hopefully @RackerWilliams will get back to us on whether this is the preferred behavior. It makes sense to me to be what we do.